core/app/models/spree/product.rb

```diff
@@ -274,6 +274,15 @@ def find_variant_property_rule(option_value_ids)
       end
     end
 
+    # Takes the product's cache key and adds a cache key for the current prices relation.
+    #
+    # This way, we expire the cache every time a new price becomes valid via its `valid_from`
+    # property.
+    #
+    def cache_key
+      super + "/prices/" + prices.valid_before_now.cache_key
```

Comment:
As I read this now, I would like to have the scopes inside of `Spree::Price.cache_key`